obspy/core/util/testing.py

```diff
@@ -706,5 +706,51 @@ def get_all_py_files():
     return sorted(py_files)
 
 
+class WarningsCapture(object):
```

Comment:
I'm a little bit confused why this monkey patching of `warnings` is even required.. but the code itself looks fine.

Reply:
Indeed, has this to do with C??

Reply:
No - its just how the warnings filtering in Python works. It is really hard to get it to behave identical across all Python versions. There are have been a number of occasions in which we could not test warnings because they did not work on all platforms.

This ctx manager just monkey patches the warnings module to get it to behave consistently. Seems to work pretty nice.